### PR TITLE
Models

### PR DESCRIPTION
**Pull request by jeremytedwards**

### CODE REVIEW

File: imager/imager_site/settings.py [hunk 1]
```diff
@@ -31,12 +31,15 @@
 # Application definition
 
 INSTALLED_APPS = [
+    'imager_profile',
+    'imager_images',
```
Comment (tlake) [inline imager/imager_site/settings.py:35]: Your own apps should be listed underneath the `django.contrib.*` stuff - otherwise, you end up needing to manually specify the apps in a specific order when you try to migrate.

File: imager/imager_images/models.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+@python_2_unicode_compatible
+class Album(models.Model):
+    """Define album class."""
+
+    owner = models.ForeignKey(settings.AUTH_USER_MODEL,
+                              on_delete=models.CASCADE,
+                              related_name="album")
+    pictures = models.ManyToManyField(Photo, related_name="pictures")
+    cover = models.ForeignKey(Photo, related_name="cover_of")
```
Comment (tlake) [inline imager/imager_images/models.py:50]: Unless you have a way to assign a value for this cover field upon creation of an Album object, this will cause DB `IntegrityError`s because you don't have `null=True` specified. This is one of the things causing your tests to fail for me.

File: imager/imager_images/models.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+@python_2_unicode_compatible
+class Photo(models.Model):
+    """Create photos for a user."""
+
+    owner = models.ForeignKey(settings.AUTH_USER_MODEL,
+                              on_delete=models.CASCADE, related_name="photo")
+    title = models.CharField(max_length=30, blank=True)
+    description = models.CharField(max_length=60, blank=True)
+    file = models.ImageField(upload_to=user_file_store_path)
```
Comment (tlake) [inline imager/imager_images/models.py:29]: You may want to rename this field so that you're not overriding the Python `file` keyword - maybe something like `img_file`?

File: imager/imager_images/tests.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        self.image2 = Photo(title='image 2', owner=self.user)
+        self.album1.save()
+        self.album2.save()
+        self.image1.save()
+        self.image2.save()
+        self.image1.albums.add(self.album1)
+
+    def test_album(self):
+        """Test that profile is created on user save."""
+        self.assertTrue(self.user.album)
```
Comment (tlake) [inline imager/imager_images/tests.py:37]: There ought to be several more tests here - in particular, having the right tests here could have helped to catch the error caused by the null value in `Album.cover`.

File: imager/imager_profile/tests.py [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+import factory
+
+
+class UserFactory(factory.django.DjangoModelFactory):
+    class Meta:
+        model = User
+
+
+class ProfileTestCase2(TestCase):
+    """Define a test case to see if testing works."""
```
Comment (tlake) [inline imager/imager_profile/tests.py:15]: Defining a test to see if testing works is not really a valuable test. If something breaks within Django's testing system, Django's tests for Django's testing system will catch those.

In general, you should only be writing tests for your own logic, to make sure that you've hooked the right wires up to the right things in the right ways. There's no need to test that Django itself works. My general rule of thumb when deciding what tests are valuable to write is: "Assume that Django works perfectly, in just the way it's supposed to. Assume that I've screwed everything up. Write tests to prove that I haven't."
